src/librustc_mir/interpret/traits.rs

```diff
@@ -32,6 +32,10 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             let trait_ref = poly_trait_ref.with_self_ty(*self.tcx, ty);
             let trait_ref = self.tcx.erase_regions(&trait_ref);
 
+            if trait_ref.needs_subst() {
+                return err!(TooGeneric);
+            }
```

Comment:
Why is this check needed here?

To me this looks like you are randomly sprinkling such checks into places. I guess there is a system, please document it.

Reply:
you are too fast. I was working on creating a clean solution that I'm not sure about :P See 2ce6d6a